src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -19,21 +19,34 @@
 
 package org.elasticsearch.script;
 
+import static org.elasticsearch.script.ScriptService.ScriptType;
+
 /**
  *
  */
 public class CompiledScript {
 
-    private final String type;
-
+    private final String lang;
+    private final String name;
+    private final ScriptType type;
```

Comment:
I added these in anticipation of #6653.  However, I'll remove them from this issue, and add them to that issue.  I'm going to leave the name change to lang.